File: Cpp/Y_Range_sum_query.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n, q;
    cin >> n >> q;

    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }

    vector<long long> prefixSum(n);
    prefixSum[0] = arr[0];
    for (int i = 1; i < n; ++i) {
        prefixSum[i] = prefixSum[i - 1] + arr[i];
    }

    while (q--) {
        int l, r;
        cin >> l >> r;

        long long sum = prefixSum[r - 1] - (l > 1 ? prefixSum[l - 2] : 0);
        cout << sum << endl;
    }

    return 0;
}

File: Python/Y_Range_sum_query.py
# def compute_prefix_sum(arr):
#     n = len(arr)
#     prefix_sum = [0] * (n+1)
#     for i in range(n):
#         prefix_sum[i+1] = prefix_sum[i] + arr[i]
#     return prefix_sum

# n, q = map(int, input().split())
# arr = list(map(int, input().split()))
# prefix_sum = compute_prefix_sum(arr)

# for _ in range(q):
#     l, r = map(int, input().split())
#     result = prefix_sum[r] - prefix_sum[l-1]
#     print(result)


n, q = map(int, input().split())
a = list(map(int, input().split()))

for _ in range(q):
    l, r = map(int, input().split())
    result = sum(a[l-1:r])
    print(result)